lib/Twig/Extensions/Extension/Filesize.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) 2015 Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * Filesize extension for human readable filesizes.
+ *
+ * @author Patrik Karisch <p.karisch@pixelart.at>
+ */
+class Twig_Extensions_Extension_Filesize extends \Twig_Extension
+{
+    protected static $two = array('Byte', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB');
+    protected static $ten = array('Byte', 'kB', 'MB', 'GB', 'TB', 'PB', 'EB');
+
+    /**
+     * The default suffix.
+     *
+     * @var string
+     */
+    protected $fixedSuffix;
+
+    /**
+     * Use power of 2 or 10 if not given in the filter.
+     *
+     * @var string
+     */
+    protected $powerOfTwo;
+
+    /**
+     * @param string $fixedSuffix Use this fixed suffix to calculate. Must be a correct
+     *                            filesize suffix and corresponding to $powerOfTwo.
+     * @param bool   $powerOfTwo  Whether to use powers of 2 or 10
+     */
+    public function __construct($fixedSuffix = null, $powerOfTwo = true)
+    {
+        $this->fixedSuffix = $fixedSuffix;
+        $this->powerOfTwo = $powerOfTwo;
+    }
+
+    /**
+     * @param string $fixedSuffix Use this fixed suffix to calculate. Must be a correct
+     *                            filesize suffix and corresponding to $powerOfTwo.
+     * @param bool   $powerOfTwo  Whether to use powers of 2 or 10
+     */
+    public function setDefaults($fixedSuffix = null, $powerOfTwo = true)
+    {
+        $this->fixedSuffix = $fixedSuffix;
+        $this->powerOfTwo = $powerOfTwo;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFilters()
+    {
+        return array(
+            new \Twig_SimpleFilter('filesize', array($this, 'filesize')),
```

Comment:
you are missing ``array('needs_env' => true)`` for the options so that it receives the Twig_Environment